ray-operator/controllers/ray/utils/util_test.go

```diff
@@ -840,3 +948,257 @@ func TestIsGCSFaultToleranceEnabled(t *testing.T) {
 		})
 	}
 }
+
+func createRayPodSpec(cpu, memory string) corev1.PodSpec {
+	return corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse(cpu),
+						corev1.ResourceMemory: resource.MustParse(memory),
+					},
+				},
+			},
+		},
+	}
+}
+
+func createRayClusterTemplate(
+	head struct {
+		cpu    string
+		memory string
+	},
+	workers []struct {
+		replicas    *int32
+		minReplicas *int32
+		suspend     *bool
+		cpu         string
+		memory      string
+		numOfHosts  int32
+	},
+) *rayv1.RayCluster {
+	cluster := &rayv1.RayCluster{
+		Spec: rayv1.RayClusterSpec{
+			HeadGroupSpec: rayv1.HeadGroupSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: createRayPodSpec(head.cpu, head.memory),
+				},
+			},
+		},
+	}
+
+	for _, w := range workers {
+		cluster.Spec.WorkerGroupSpecs = append(cluster.Spec.WorkerGroupSpecs, rayv1.WorkerGroupSpec{
+			NumOfHosts:  w.numOfHosts,
+			Replicas:    w.replicas,
+			MinReplicas: w.minReplicas,
+			Suspend:     w.suspend,
+			Template: corev1.PodTemplateSpec{
+				Spec: createRayPodSpec(w.cpu, w.memory),
+			},
+		})
+	}
+
+	return cluster
+}
+
+func TestCalculateDesiredResources(t *testing.T) {
```

Comment:
can we fuse `TestCalculateDesiredResources` and `TestCalculateMinResources` together? There are many duplicate code.